libraries/lib-ffmpeg-support/wrappers/AVCodecContextWrapper.cpp

```diff
@@ -79,24 +79,39 @@ AVCodecContextWrapper::DecodeAudioPacket(const AVPacketWrapper* packet)
       std::unique_ptr<AVPacketWrapper> packetCopy =
          packet ? packet->Clone() : mFFmpeg->CreateAVPacketWrapper();
 
-      int packetSizeLeft = packet ? packetCopy->GetSize() : 0;
-
-      if (!packetSizeLeft)
-         packetSizeLeft = 1;
+      /*
+       "Flushing is done by calling this function [avcodec_decode_audio4]
+       with packets with avpkt->data set to NULL and avpkt->size set to 0
+       until it stops returning samples."
+       (That implies, not necessarily just one loop pass to flush)
+       */
+      bool flushing = packet
+         ? (packetCopy->GetSize() == 0 && packetCopy->GetData() == nullptr)
+         : true;
+      if (!flushing && packetCopy->GetData() == nullptr)
+         return {};
 
       const int channels = GetChannels();
 
-      while (packetSizeLeft > 0)
+      int bytesDecoded = 0;
+      do
       {
          int gotFrame;
-         const int bytesDecoded = mFFmpeg->avcodec_decode_audio4(
+         // Deprecated?  https://ffmpeg.org/doxygen/3.3/group__lavc__decoding.html#gaaa1fbe477c04455cdc7a994090100db4
```

Comment:
Yes, but new API is not available in 2.2.2 and if we end up with FFmpeg version without avcodec_decode_audio4 - it can be implemented using new API